include/bluetooth/bluetooth.h

```diff
@@ -71,6 +79,36 @@ int bt_enable(bt_ready_cb_t cb);
  */
 int bt_set_id_addr(const bt_addr_le_t *addr);
 
+/** @brief Get the currently configured identities.
+ *
+ *  Returns an array of the currently configured identity addresses.
+ *
+ *  @param addrs Array where to store the configured identities.
+ *  @param count Should be initialized to the array size. Once the function
+ *               returns it will contain the number of returned identies.
+ */
+void bt_id_get(bt_addr_le_t *addrs, size_t *count);
+
+/** @brief Create a new identity.
+ *
+ *  Create a new identity using the given address and IRK.
+ *
+ *  @param addr Address to use for the new identity. If initialized to
+ *              BT_ADDR_LE_ANY the stack will generate a new static
+ *              random address for the identity and copy it to the given
+ *              parameter upon return from this function.
+ *  @param irk  Identity Resolving Key (16 bytes) to be used with this
+ *              identity. If set to all zeroes or NULL, the stack will
+ *              generate a random IRK for the identity and copy it back
+ *              to the parameter upon return from this function. If
+ *              privacy support (CONFIG_BT_PRIVACY) is not enabled this
+ *              parameter must be NULL.
+ *
+ *  @return Identity identifier (>= 0) in case of success, or a negative
+ *          error code on failure.
+ */
+int bt_id_create(bt_addr_le_t *addr, u8_t *irk);
```

Comment:
How we can delete identity?

Reply:
In the github issue that tracks this (#7473) there seemed to be a consensus that this would not be a safe API to have. So the only way to delete stored identities is to erase the flash storage area and do a power cycle.

Reply:
I don't think that it is acceptable solution. Please look for use cases for multiple identities:
https://github.com/zephyrproject-rtos/zephyr/issues/7473#issuecomment-388741527

We definitely must have a method for deleting (or revoking and later recycling) given identity.

Reply:
With `bt_id_reset()` that I added you can both revoke and recycle. To revoke you'd simply issue `bt_unpair(id, NULL);` and to recycle you'd call `bt_id_reset()`, or in a single step just call `bt_id_reset()`.